src/dataset-builder/ConceptSearch.ts

```diff
@@ -33,15 +34,24 @@ const DebouncedTextInput = withDebouncedChange(TextInput);
 export const ConceptSearch = (props: ConceptSearchProps) => {
   const { initialSearch, domainOption, onCancel, onCommit, onOpenHierarchy, actionText, datasetId, initialCart } =
     props;
-  const [search, setSearch] = useState<string>(initialSearch);
+  const [searchText, setSearchText] = useState<string>(initialSearch);
   const [cart, setCart] = useState<Concept[]>(initialCart);
   const [concepts, searchConcepts] = useLoadedData<GetConceptsResponse>();
 
-  useEffect(() => {
+  // when mounting ConceptSearch, we want to retrieve a list of searchConcepts
+  useOnMount(() => {
     void searchConcepts(() => {
-      return DataRepo().dataset(datasetId).searchConcepts(domainOption.root, search);
+      // if searchText is '' then searchConcepts will error out, we must use ' '
+      return DataRepo().dataset(datasetId).searchConcepts(domainOption.root, ' ');
```

Comment:
So I think this means by default we are only showing concepts that have spaces in them. That seems like a strange default to me. What do you think? If we can't pass the empty string, do we need a new endpoint to just get all concepts?

Reply:
How would you feel if we rather changed `searchConcepts` to support the empty string? It should just remove the where clause on an empty string.

Reply:
I like the idea of modifying searchConcepts to be able to handle an empty string!

Reply:
Good idea!

Reply:
I put in a small PR for this change: https://github.com/DataBiosphere/jade-data-repo/pull/1600/files !

Reply:
great work! I'm checking this out :)